include/rmm/cuda_stream_pool.hpp

```diff
@@ -61,6 +62,33 @@ class cuda_stream_pool {
     return streams_[(next_stream++) % streams_.size()].view();
   }
 
+  /**
+   * @brief Get a `cuda_stream_view` of the stream at stream_id index the pool.
+   *
+   * This function is thread safe with respect to other calls to the same function.
+   *
+   * @throws rmm::out_of_range exception if `stream_index >= size()`
+   *
+   * @param stream_id The index of the stream in the pool
+   *
+   * @return rmm::cuda_stream_view
+   */
+  rmm::cuda_stream_view get_stream(std::size_t stream_index) const
+  {
+    RMM_EXPECTS(
+      stream_index < streams_.size(), rmm::out_of_range, "Attempt to access out of bounds stream.");
```

Comment:
Related to my above comments about treating the argument as a unique identifier instead of a literal index, we should just do:
```
return streams_[stream_id % streams_.size()].view();
```

Reply:
898cf5d